FhirToDataLake/src/Microsoft.Health.Fhir.Synapse.Core/Jobs/AzureBlobJobStore.cs

```diff
@@ -50,23 +54,95 @@ public class AzureBlobJobStore : IJobStore
 
         public AzureBlobJobStore(
             IAzureBlobContainerClientFactory blobContainerFactory,
+            IFhirSpecificationProvider fhirSpecificationProvider,
             IOptions<JobConfiguration> jobConfiguration,
             IOptions<DataLakeStoreConfiguration> storeConfiguration,
             ILogger<AzureBlobJobStore> logger)
         {
             EnsureArg.IsNotNull(blobContainerFactory, nameof(blobContainerFactory));
+            EnsureArg.IsNotNull(fhirSpecificationProvider, nameof(fhirSpecificationProvider));
             EnsureArg.IsNotNull(jobConfiguration, nameof(jobConfiguration));
             EnsureArg.IsNotNull(storeConfiguration, nameof(storeConfiguration));
             EnsureArg.IsNotNull(logger, nameof(logger));
 
-            _blobContainerClient = blobContainerFactory.Create(storeConfiguration.Value.StorageUrl, jobConfiguration.Value.ContainerName);
+            _fhirSpecificationProvider = fhirSpecificationProvider;
+            _jobConfiguration = jobConfiguration.Value;
+            _blobContainerClient = blobContainerFactory.Create(storeConfiguration.Value.StorageUrl, _jobConfiguration.ContainerName);
             _logger = logger;
 
             _renewLockTimer = new Timer(TimeSpan.FromSeconds(AzureBlobJobConstants.JobLeaseRefreshIntervalInSeconds).TotalMilliseconds);
             _renewLockTimer.Elapsed += async (sender, e) => await RenewJobLockLeaseAsync();
         }
 
-        public async Task<bool> AcquireJobLock(CancellationToken cancellationToken = default)
+        public async Task<Job> AcquireJobAsync(CancellationToken cancellationToken = default)
+        {
+            var lockAcquired = await TryAcquireJobLock(cancellationToken);
+            if (!lockAcquired)
+            {
+                _logger.LogWarning("Start job conflicted. Failed to acquire job lock.");
+                throw new StartJobFailedException("Start job conflicted. Will skip this trigger.");
+            }
+
+            Job job = null;
+            var activeJobs = await GetActiveJobsAsync(cancellationToken);
+            if (activeJobs.Any())
+            {
+                // Resume an active job.
+                job = activeJobs.First();
+
+                if (job.Status == JobStatus.Succeeded)
+                {
+                    _logger.LogWarning("Job '{id}' has already succeeded.", job.Id);
+                    await CompleteJobInternal(job, false, cancellationToken);
+                    job = null;
+                }
+                else
+                {
+                    // Resume an inactive/failed job.
+                    job.Status = JobStatus.Running;
+                    job.FailedReason = null;
+                }
+            }
+
+            // No active job available, start new trigger.
+            if (job == null)
+            {
+                job = await CreateNewJob(cancellationToken);
+            }
```

Comment:
It looks to me that these logics might belong to JobManager (create new, change status, etc.) instead of JobStore?

Reply:
Good idea. Moved the control logic to job manager